backend/http/http.go

```diff
@@ -365,6 +378,36 @@ func parse(base *url.URL, in io.Reader) (names []string, err error) {
 	return names, nil
 }
 
+type GoIndexFile struct {
+	Id           string `json:"id"`
+	MimeType     string `json:"mimeType"`
+	ModifiedTime string `json:"modifiedTime"`
+	Name         string `json:"name"`
+	// don't know why, but this is a string instead of an int64 (against gdrive doc)
+	Size          string `json:"size"`
+	ThumbnailLink string `json:"thumbnailLink"`
+}
+
+type GoIndexResults struct {
+	Data struct {
+		Files []GoIndexFile `json:"files"`
+	} `json:"data"`
+}
+
+// parser for GoIndex remotes
+func parseGoIndex(u *url.URL, in io.Reader) (names []string, err error) {
+	var results GoIndexResults
+	dec := json.NewDecoder(in)
+	err = dec.Decode(&results)
+	if err != nil {
+		return nil, err
+	}
+	for _, goIndexFile := range results.Data.Files {
+		names = append(names, goIndexFile.Name)
+	}
+	return names, nil
```

Comment:
@ncw So this is where we get into "should this be its own backend territory", because

* I can modify the http backend to return `Object`s here and in the normal `parse`, adding a name field, and modify the rest of the `List` routine to handle that. Don't know if there's significantly more cost to passing around `Object`s with empty fields bar "Name" (in the case of normal http remotes). But doing this would provide a significant speed boost to GoIndex remotes, because instead of doing a POST for each file, stats can be determined from the directory listing (at least initially)
*  Copy+paste a lot of the http backend code into a new, "goindex" backend, so violating DRY, but avoiding and GoIndex specifics in the http backend.
* While still unimplemented by me, the GdIndex flavor supports uploads (still figuring that out)
* While still unimplemented by me, GoIndexes support directory-level (afaik piercing, but not general) passwords *on top of* HTTP basic authentication. Ex, without the password, if someone knows a file in the directory they still can *read* it as well as *stat* it, but the user can't list the directory (ex, on *nix, mark the directory with `chmod u-r+x`

I'm fine with either keeping this in http or making it separate, just that I'd want this to be Nick's / maintainers' decision, not mine. I would imagine a different backend would be preferable, because otherwise one could just keep adding tiny hack upon tiny hack for each specific custom http remote to the http backend until it becomes a gigantic mess.